kentikapi/models/cloud_export.go

```diff
@@ -0,0 +1,239 @@
+			Subscription: obj.GCEProperties.Subscription,
+		},
+	}
+}
+
+// NewIBMCloudExport creates a new IBM CloudExport with all required fields set.
+func NewIBMCloudExport(obj CloudExportIBMRequiredFields) *CloudExport {
+	return &CloudExport{
+		Name:          obj.Name,
+		PlanID:        obj.PlanID,
+		CloudProvider: CloudProviderIBM,
+		IBMProperties: &IBMProperties{
+			Bucket: obj.IBMProperties.Bucket,
+		},
+	}
+}
+
+// GetAllCloudExportsResponse model.
+type GetAllCloudExportsResponse struct {
+	// CloudExports holds all cloud export tasks.
+	CloudExports []CloudExport
+	// InvalidCloudExportsCount is a number of invalid cloud export tasks.
+	InvalidCloudExportsCount uint32
+}
+
+// CloudExport defines a cloud export task.
+type CloudExport struct {
+	// Read-write properties
+
+	// Type of export task.
+	Type CloudExportType
+	// Enabled specifies whether this task is enabled and intended to run or disabled.
+	Enabled *bool
+	// Name is a short name for this export task.
+	Name string
+	// Description is an optional, longer description of this export task.
+	Description string
+	// APIRoot is a hostname of the Kentik deployment where the API calls related to this export should go,
+	// e.g. "api.kentik.com".
+	APIRoot string
+	// FlowDestination is a hostname of the Kentik deployment where the data generated by this export should go,
+	// e.g. "flow.kentik.com".
+	FlowDestination string
+	// PlanID is the identifier of the Kentik plan associated with this export.
+	PlanID string
+	// CloudProvider is the cloud provider targeted by this export, e.g. AWS, Azure, GCE, IBM.
+	CloudProvider CloudProvider
+	// AWSProperties are specific to Amazon Web Services VPC flow logs exports.
+	AWSProperties *AWSProperties
+	// AzureProperties are specific to Azure exports.
+	AzureProperties *AzureProperties
+	// GCEProperties are specific to Google Cloud export.
+	GCEProperties *GCEProperties
+	// IBMProperties are specific to IBM Cloud exports.
+	IBMProperties *IBMProperties
+	// BGPProperties are optional BGP related settings.
+	BGP *BGPProperties
+
+	// Read-only properties
+
+	// ID is the internal cloud export identifier. This is read-only and assigned by Kentik.
+	ID ID
+	// CurrentStatus is the most current status Kentik has about this export. This is read-only and assigned by Kentik.
+	CurrentStatus *CloudExportStatus
+}
+
+// AWSProperties are specific to Amazon Web Services VPC flow logs exports.
+type AWSProperties struct {
+	// Bucket is source S3 bucket to fetch VPC flow logs from.
+	Bucket string
+	// IAMRoleARN is an ARN for the IAM role to assume when fetching data or making AWS calls for this export.
+	IAMRoleARN string
+	// Region is an AWS region where the source S3 bucket resides.
+	Region string
+	// DeleteAfterRead enables the deletion of VPC flow log chunks from S3 after they have been read.
+	DeleteAfterRead *bool
+	// MultipleBuckets enables using multiple source S3 buckets.
+	MultipleBuckets *bool
```

Comment:
This will also be changed (or possibly removed), because it actually does not mean what the name of the field implies.

Reply:
Should I remove it now?